### PR TITLE
feat(todo-13): create API route to add a task

### PR DESCRIPTION
**Pull request by syed-alif-sazim**
**Description of change**
An API route is created to add a task in the database

**Associated trello link**
https://trello.com/c/F8jOfYR3

### CODE REVIEW

File: src/task/task.controller.ts [hunk 1]
```diff
@@ -10,4 +10,9 @@ export class TaskController {
   async getAllTasks(): Promise<Task[]> {
     return this.taskService.getAllTasks();
   }
+
+  @Post()
+  async create(@Body() newTaskObj: { description: string; isCompleted: boolean }) {
```
Comment (mahir-mohsin-sazim) [inline src/task/task.controller.ts:15]: use DTOs to validate payload in the backend.

File: src/task/task.service.ts [hunk 2]
```diff
@@ -13,4 +13,15 @@ export class TaskService {
   async getAllTasks(): Promise<Task[]> {
     return this.taskRepository.findAll();
   }
+
+  async createTask(newTaskObj: { description: string; isCompleted: boolean }): Promise<Task> {
```
Comment (mahir-mohsin-sazim) [inline src/task/task.service.ts:17]: same here

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]

File: src/task/task.controller.ts (hunk 1 above)
Comment (syed-alif-sazim) [reply src/task/task.controller.ts:15]: I resolved this in the last PR

File: src/task/task.service.ts (hunk 2 above)
Comment (syed-alif-sazim) [reply src/task/task.service.ts:17]: I resolved this in the last PR

**Review by mahir-mohsin-sazim — APPROVED** [review]

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**